rct229/rulesets/ashrae9012019/section5/section5rule38.py

```diff
@@ -1,163 +1,51 @@
+            description="It is acceptable to use either an annual average ground temperature or monthly average ground temperatures for calculation of heat loss through below-grade walls and basement floors.",
             ruleset_section_title="Envelope",
-            standard_section="Section G3.1-5(b) Building Envelope Modeling Requirements for the Proposed design",
-            is_primary_rule=True,
-            list_path="ruleset_model_descriptions[0].buildings[*]",
-            data_items={"climate_zone": (PROPOSED, "weather/climate_zone")},
-            manual_check_required_msg=MANUAL_CHECK_MSG,
+            standard_section="Section G3.1-14(b) Building Envelope Modeling Requirements for the Proposed design and Baseline",
+            is_primary_rule=False,
+            required_fields={
+                "$": ["weather", "ruleset_model_descriptions"],
+            },
         )
 
-    class BuildingRule(RuleDefinitionBase):
-        def __init__(self):
-            super(Section5Rule38.BuildingRule, self).__init__(
-                rmrs_used=produce_ruleset_model_instance(
-                    USER=False, BASELINE_0=False, PROPOSED=True
-                ),
-                required_fields={"$..zones[*]": ["surfaces"]},
-            )
-
-        def get_calc_vals(self, context, data=None):
-            building_p = context.PROPOSED
-
-            scc_dict_p = get_surface_conditioning_category_dict(
-                data["climate_zone"], building_p
-            )
-            zcc_dict_p = get_zone_conditioning_category_dict(
-                data["climate_zone"], building_p
-            )
-
-            building_total_air_leakage_rate = ZERO.FLOW
-            building_total_measured_air_leakage_rate = ZERO.FLOW
-            empty_measured_air_leakage_rate_flow_flag = False
-
-            building_total_envelope_area = sum(
-                [
-                    getattr_(surface, "surface", "area")
-                    for surface in find_all("$..surfaces[*]", building_p)
-                    if scc_dict_p[surface["id"]] != SCC.UNREGULATED
-                ],
-                ZERO.AREA,
-            )
-
-            for zone in find_all("$..zones[*]", building_p):
-                if zcc_dict_p[zone["id"]] in [
-                    ZCC.CONDITIONED_RESIDENTIAL,
-                    ZCC.CONDITIONED_NON_RESIDENTIAL,
-                    ZCC.CONDITIONED_MIXED,
-                    ZCC.SEMI_HEATED,
-                ]:
-                    building_total_air_leakage_rate += getattr_(
-                        zone["infiltration"], "infiltration", "flow_rate"
-                    )
-
-                measured_air_leakage_rate = zone["infiltration"].get(
-                    "measured_air_leakage_rate"
+    def is_applicable(self, context, data=None):
+        rpd = context.BASELINE_0
+        return any(
+            [
+                surface_b
```

Comment:
I think you can do more explicit logic here:
```python
any([get_opaque_surface_type(surface_b) == OST.BELOW_GRADE_WALL for surface_b in find_all(..., rpd)])
```

Reply:
Good catch!